CMake/SiloFindQt5.cmake

```diff
@@ -0,0 +1,107 @@
+#      notice, this list of conditions and the disclaimer below.
+#    * Redistributions in binary form must reproduce the above copyright
+#      notice, this list of conditions and the disclaimer (as noted
+#      below) in the documentation and/or other materials provided with
+#      the distribution.
+#    * Neither the name of the LLNS/LLNL nor the names of its
+#      contributors may be used to endorse or promote products derived
+#      from this software without specific prior written permission.
+# 
+# THIS SOFTWARE  IS PROVIDED BY  THE COPYRIGHT HOLDERS  AND CONTRIBUTORS
+# "AS  IS" AND  ANY EXPRESS  OR IMPLIED  WARRANTIES, INCLUDING,  BUT NOT
+# LIMITED TO, THE IMPLIED  WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A  PARTICULAR  PURPOSE ARE  DISCLAIMED.  IN  NO  EVENT SHALL  LAWRENCE
+# LIVERMORE  NATIONAL SECURITY, LLC,  THE U.S.  DEPARTMENT OF  ENERGY OR
+# CONTRIBUTORS BE LIABLE FOR  ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR  CONSEQUENTIAL DAMAGES  (INCLUDING, BUT NOT  LIMITED TO,
+# PROCUREMENT OF  SUBSTITUTE GOODS  OR SERVICES; LOSS  OF USE,  DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+# LIABILITY, WHETHER  IN CONTRACT, STRICT LIABILITY,  OR TORT (INCLUDING
+# NEGLIGENCE OR  OTHERWISE) ARISING IN  ANY WAY OUT  OF THE USE  OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# 
+# This work was produced at Lawrence Livermore National Laboratory under
+# Contract No.  DE-AC52-07NA27344 with the DOE.
+# 
+# Neither the  United States Government nor  Lawrence Livermore National
+# Security, LLC nor any of  their employees, makes any warranty, express
+# or  implied,  or  assumes  any  liability or  responsibility  for  the
+# accuracy, completeness,  or usefulness of  any information, apparatus,
+# product, or  process disclosed, or  represents that its use  would not
+# infringe privately-owned rights.
+# 
+# Any reference herein to  any specific commercial products, process, or
+# services by trade name,  trademark, manufacturer or otherwise does not
+# necessarily  constitute or imply  its endorsement,  recommendation, or
+# favoring  by  the  United  States  Government  or  Lawrence  Livermore
+# National Security,  LLC. The views  and opinions of  authors expressed
+# herein do not necessarily state  or reflect those of the United States
+# Government or Lawrence Livermore National Security, LLC, and shall not
+# be used for advertising or product endorsement purposes.
+#
+##############################################################################
+
+###
+# Uses find_package to find the Qt5 library
+#   If SILO_QT5_DIR is defined, uses it to tell CMake where to look.
+###
+
+
+if(DEFINED SILO_QT5_DIR AND EXISTS ${SILO_QT5_DIR})
+    # set the variable needed for CMake to find this version of Qt
+    set(Qt5_DIR ${SILO_QT5_DIR})
+endif()
+
+find_package(Qt5 COMPONENTS Core Gui Widgets CONFIG)
```

Comment:
Do you happen to know if this finding logic works against a VisIt-managed third_party_shared install of Qt? That is ultimately what I've wound up using to build Silo on LLNL systems and I just wanna make sure that will still work. I have had no luck finding *system* Qt version 5. Only version 3 (like from a hundred years ago)

Reply:
I built silo on pascal without setting SILO_QT5_DIR and CMake found system Qt5 just fine, and silex built correctly and runs. 

These are the QT env vars I unset so that they don't interfere with the Qt I am building or want to use:
unsetenv QTDIR
unsetenv QTINC
unsetenv QTLIB
unsetenv QT_PLUGIN_PATH